Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeSwitcher.cpp

```diff
@@ -0,0 +1,381 @@
+            m_iconPath,
+            &AzToolsFramework::EditorRequestBus::Events::GetComponentEditorIcon,
+            m_component->GetUnderlyingComponentType(),
+            m_component);
+    }
+
+    ComponentModeSwitcher::ComponentModeSwitcher()
+    {
+        ViewportEditorModeNotificationsBus::Handler::BusConnect(GetEntityContextId());
+        EntityCompositionNotificationBus::Handler::BusConnect();
+        ToolsApplicationNotificationBus::Handler::BusConnect();
+
+
+        // Create the switcher
+        ViewportUi::ViewportUiRequestBus::EventResult(
+            m_switcherId,
+            ViewportUi::DefaultViewportId,
+            &ViewportUi::ViewportUiRequestBus::Events::CreateSwitcher,
+            ViewportUi::Alignment::TopLeft);
+
+        // Initial Transform button
+        ViewportUi::ViewportUiRequestBus::EventResult(
+            m_transformButtonId,
+            ViewportUi::DefaultViewportId,
+            &ViewportUi::ViewportUiRequestBus::Events::CreateSwitcherButton,
+            m_switcherId,
+            "../../../../Assets/Editor/Icons/Components/Transform.svg",
+            "Transform");
+
+        ViewportUi::ViewportUiRequestBus::Event(
+            ViewportUi::DefaultViewportId,
+            &ViewportUi::ViewportUiRequestBus::Events::SetSwitcherActiveButton,
+            m_switcherId,
+            m_transformButtonId);
+
+        // Handler
+        auto handlerFunc = [this](ViewportUi::ButtonId buttonId)
+        {
+            ViewportUi::ViewportUiRequestBus::Event(
+                ViewportUi::DefaultViewportId,
+                &ViewportUi::ViewportUiRequestBus::Events::SetSwitcherActiveButton,
+                m_switcherId,
+                buttonId);
+
+            ActivateComponentMode(buttonId);
+        };
+
+        m_handler = AZ::Event<ViewportUi::ButtonId>::Handler(handlerFunc);
+        // Registers the SwitcherEventHandler which is defined in ComponentModeSwitcher
+        ViewportUi::ViewportUiRequestBus::Event(
+            ViewportUi::DefaultViewportId,
+            &ViewportUi::ViewportUiRequestBus::Events::RegisterSwitcherEventHandler,
+            m_switcherId,
+            m_handler);
+    }
+
+    ComponentModeSwitcher::~ComponentModeSwitcher()
+    {
+        ViewportEditorModeNotificationsBus::Handler::BusDisconnect();
+        EntityCompositionNotificationBus::Handler::BusDisconnect();
+        ToolsApplicationNotificationBus::Handler::BusDisconnect();
+
+        ViewportUi::ViewportUiRequestBus::Event(
+            ViewportUi::DefaultViewportId, &ViewportUi::ViewportUiRequestBus::Events::RemoveSwitcher, m_switcherId);
+    }
+
+    void ComponentModeSwitcher::UpdateSwitcherOnEntitySelectionChange(
+        [[maybe_unused]] const EntityIdList newlySelectedEntityIds, [[maybe_unused]] const EntityIdList newlyDeselectedEntityIds)
+    {
+        AZ::Entity* entity = nullptr;
+
+        // Currently only handling one new entity selection at a time
```

Comment:
question: Is this comment still valid? I thought we can now correctly handle more than one entity being selected?

Reply:
Technically no, we can handle multiple entities yes but not if they're selected at the same time, how would you do this in the editor?

Reply:
but the comment is still wrong in this case

Reply:
Okay cool thanks for reviewing and updating :+1: